src/mpi/coll/algorithms/stub/impl.h

```diff
@@ -0,0 +1,447 @@
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#include <stdio.h>
+#include <string.h>
+
+#ifndef MPIR_ALGO_NAMESPACE
+#error "The collectives template must be namespaced with MPIR_ALGO_NAMESPACE"
+#endif
+
+
+
+MPL_STATIC_INLINE_PREFIX int MPIR_ALGO_init()
+{
+    return 0;
+}
+
+
+MPL_STATIC_INLINE_PREFIX int MPIR_ALGO_comm_init(MPIR_ALGO_comm_t * comm, int *tag, int rank, int size)
+{
+    MPIR_TSP_comm_init(&comm->tsp_comm, MPIR_ALGO_COMM_BASE(comm));
+    return 0;
+}
+
+MPL_STATIC_INLINE_PREFIX int MPIR_ALGO_comm_init_null(MPIR_ALGO_comm_t * comm)
+{
+    MPIR_TSP_comm_init_null(&comm->tsp_comm, MPIR_ALGO_COMM_BASE(comm));
+    return 0;
+}
+
+MPL_STATIC_INLINE_PREFIX int MPIR_ALGO_comm_cleanup(MPIR_ALGO_comm_t * comm)
+{
+    MPIR_TSP_comm_cleanup(&comm->tsp_comm);
+    return 0;
+}
+
+MPL_UNUSED MPL_STATIC_INLINE_PREFIX int MPIR_ALGO_allgather(const void *sendbuf, int sendcount, MPIR_ALGO_dt_t sendtype,
+                               void *recvbuf, int recvcount, MPIR_ALGO_dt_t recvtype, MPIR_ALGO_comm_t * comm,
+                               int k, int halving, int *errflag)
+{
+    MPIR_COLL_Assert(0);
```

Comment:
Just use the regular `MPIR_Assert` function throughout. The `MPIR_COLL_Assert` function should be removed.

Reply:
Done